spack.yaml

```
@@ -4,58 +4,63 @@
 # configuration settings.
 spack:
   specs:
-    # TODO: Replace the MODEL and VERSION.
-    # The root SBD for the model and overall version of the deployment:
-    # - MODEL@git.VERSION
+    - access-am3@git.2025.06.000
   packages:
-    # TODO: Specify versions and variants of dependencies where required
-    # Specification of dependency versions and variants goes here.
-    # CI/CD requires that the first element of the require is only a version:
-    # openmpi:
-    #   require:
-    #     - '@4.0.2'
+    # Direct dependencies
+    um:
+      require:
+        - '@13.1'
+        - model="vn13p1-am"
+        - jules_sources="AM3-dev"
+        - um_sources="AM3-dev"
+
+    # Indirect dependencies
+    eccodes:
+      require:
+         - '@2.34.0'
+
+    netcdf-c:
+      require:
+        - '@4.9.2'
+
+    netcdf-fortran:
+      require:
+        - '@4.5.2'
+
+    fcm:
+      require:
+        - '@2021.05.0'
+        # TODO: Generalize this Gadi-specific variant for spack.yaml
+        - 'site=nci-gadi'
+
+    gcom:
+      require:
+        - '@7.9'
+
+    openmpi:
+      require:
+        - '@5.0.5'
 
     # Specifications that apply to all packages
     all:
-      # TODO: Specify compiler/targets for all packages
-      # require:
-      #   - '%intel@19.0.5.281'
-      #   - 'target=x86_64'
+      require:
+        - '%intel@2021.10.0'
+        - target=x86_64_v4
   view: true
   concretizer:
     unify: true
   modules:
     default:
-      enable:
-        - tcl
-      roots:
-        tcl: $spack/../release/modules
-        lmod: $spack/../release/lmod
       tcl:
-        hash_length: 0
         include:
-          # Explicitly, which packages are accessible as modules
-          # TODO: Add packages that will be included as modules
-          # - MODEL
-        exclude_implicits: true
-        all:
-          autoload: direct
-          conflict:
-            - '{name}'
-          environment:
-            set:
-              'SPACK_{name}_ROOT': '{prefix}'
+          - access-am3
+          - um
```

```suggestion
          - um
          - openmpi
```
This will load the correct `openmpi` library with the spack executable module, meaning it doesn't require a hard-coded `module load openmpi/XXXX` in the suite. This is highly desirable as it makes the suite more generic and does not require changes in the suite when the openmpi version is changed.

See https://github.com/ACCESS-NRI/ACCESS-TEST/pull/49 for tests.

---

This seems not to have worked. I suspect the issue is the `um` package does not directly depend on `mpi`, which is an error that needs to be fixed. See related issue https://github.com/ACCESS-NRI/spack-packages/issues/293